### PR TITLE
reinstate phpcs php linting

### LINKED ISSUE #1356: Add PHPCS to Storefront
To improve the code quality of the Storefront theme, I propose adding the adding the following tools:

- https://github.com/squizlabs/PHP_CodeSniffer
- https://github.com/WordPress/WordPress-Coding-Standards
- https://github.com/WPTRT/WPThemeReview

**Motivation**

When running `phpcs . --standard=WordPress,WPThemeReview --ignore="/vendor/*,/node_modules/*" --extensions=php` within the theme, the following errors and warnings are listed:

```
FILE: /Users/nielslange/Development/Themes/storefront/404.php
-----------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AND 1 WARNING AFFECTING 3 LINES
-----------------------------------------------------------------------------------------------------------------------
 50 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 64 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 70 | WARNING | [ ] Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore
    |         |     Standard.Category.SniffName.ErrorCode" annotations instead. Found: // WPCS: XSS ok.
    |         |     
-----------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------


FILE: /Users/nielslange/Development/Themes/storefront/inc/wordpress-shims.php
-----------------------------------------------------------------------------
FOUND 11 ERRORS AFFECTING 11 LINES
-----------------------------------------------------------------------------
  9 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
 10 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
 11 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
 12 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
 13 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
 14 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
 15 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
 16 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
 17 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
 18 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed
 19 | ERROR | [x] File must end with a newline character
-----------------------------------------------------------------------------
PHPCBF CAN FIX THE 11 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------


FILE: /Users/nielslange/Development/Themes/storefront/inc/woocommerce/class-storefront-woocommerce-customizer.php
-----------------------------------------------------------------------------------------------------------------
FOUND 5 ERRORS AFFECTING 5 LINES
-----------------------------------------------------------------------------------------------------------------
 58 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 66 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 73 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 80 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 90 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
-----------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 5 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------


FILE: /Users/nielslange/Development/Themes/storefront/inc/woocommerce/storefront-woocommerce-template-functions.php
-----------------------------------------------------------------------------------------------------------------------
FOUND 26 ERRORS AND 11 WARNINGS AFFECTING 37 LINES
-----------------------------------------------------------------------------------------------------------------------
 254 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 264 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 265 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 286 | WARNING | [ ] Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore
     |         |     Standard.Category.SniffName.ErrorCode" annotations instead. Found: // WPCS: XSS ok.
     |         |     
 306 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 316 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 317 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 338 | WARNING | [ ] Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore
     |         |     Standard.Category.SniffName.ErrorCode" annotations instead. Found: // WPCS: XSS ok.
     |         |     
 358 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 369 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 370 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 392 | WARNING | [ ] Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore
     |         |     Standard.Category.SniffName.ErrorCode" annotations instead. Found: // WPCS: XSS ok.
     |         |     
 412 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 422 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 423 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 444 | WARNING | [ ] Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore
     |         |     Standard.Category.SniffName.ErrorCode" annotations instead. Found: // WPCS: XSS ok.
     |         |     
 464 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 475 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 476 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 498 | WARNING | [ ] Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore
     |         |     Standard.Category.SniffName.ErrorCode" annotations instead. Found: // WPCS: XSS ok.
     |         |     
 518 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 528 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 529 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 550 | WARNING | [ ] Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore
     |         |     Standard.Category.SniffName.ErrorCode" annotations instead. Found: // WPCS: XSS ok.
     |         |     
 583 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 587 | WARNING | [ ] Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore
     |         |     Standard.Category.SniffName.ErrorCode" annotations instead. Found: // WPCS: XSS ok.
     |         |     
 593 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 597 | WARNING | [ ] Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore
     |         |     Standard.Category.SniffName.ErrorCode" annotations instead. Found: // WPCS: XSS ok.
     |         |     
 603 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 607 | WARNING | [ ] Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore
     |         |     Standard.Category.SniffName.ErrorCode" annotations instead. Found: // WPCS: XSS ok.
     |         |     
 768 | WARNING | [x] Usage of ELSE IF is discouraged; use ELSEIF instead
 777 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 821 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 831 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 832 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 849 | WARNING | [ ] Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore
     |         |     Standard.Category.SniffName.ErrorCode" annotations instead. Found: // WPCS: XSS ok.
     |         |     
 890 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
-----------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 27 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------


FILE: /Users/nielslange/Development/Themes/storefront/inc/woocommerce/class-storefront-woocommerce.php
------------------------------------------------------------------------------------------------------
FOUND 3 ERRORS AND 3 WARNINGS AFFECTING 6 LINES
------------------------------------------------------------------------------------------------------
  60 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
  61 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 191 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 416 | WARNING | [ ] Found precision alignment of 1 spaces.
 417 | WARNING | [ ] Found precision alignment of 1 spaces.
 418 | WARNING | [ ] Found precision alignment of 1 spaces.
------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------------------------------------


FILE: ...Users/nielslange/Development/Themes/storefront/inc/woocommerce/class-storefront-woocommerce-adjacent-products.php
-----------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
-----------------------------------------------------------------------------------------------------------------------
 87 | WARNING | Variable assignment found within a condition. Did you mean to do a comparison?
-----------------------------------------------------------------------------------------------------------------------


FILE: /Users/nielslange/Development/Themes/storefront/inc/storefront-template-functions.php
-----------------------------------------------------------------------------------------------------------------------
FOUND 3 ERRORS AND 4 WARNINGS AFFECTING 6 LINES
-----------------------------------------------------------------------------------------------------------------------
  52 | ERROR   | [ ] The default value of the $encoding parameter for htmlspecialchars() was changed from ISO-8859-1
     |         |     to UTF-8 in PHP 5.4. For cross-version compatibility, the $encoding parameter should be
     |         |     explicitly set.
  53 | WARNING | [ ] All output should be run through an escaping function (see the Security sections in the WordPress
     |         |     Developer Handbooks), found 'get_comment_date'.
  53 | WARNING | [ ] All output should be run through an escaping function (see the Security sections in the WordPress
     |         |     Developer Handbooks), found 'get_comment_date'.
  66 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 148 | WARNING | [x] Equals sign not aligned with surrounding assignments; expected 4 spaces but found 1 space
 227 | WARNING | [ ] Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore
     |         |     Standard.Category.SniffName.ErrorCode" annotations instead. Found: // WPCS: XSS ok.
     |         |     
 492 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
-----------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------


FILE: /Users/nielslange/Development/Themes/storefront/inc/customizer/class-storefront-customizer.php
-----------------------------------------------------------------------------------------------------------------------
FOUND 65 ERRORS AND 11 WARNINGS AFFECTING 57 LINES
-----------------------------------------------------------------------------------------------------------------------
  42 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 135 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 142 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 149 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 159 | WARNING | [ ] Check that require_once is not being used to load template files. "get_template_part()" should be
     |         |     used to load template files.
 160 | WARNING | [ ] Check that require_once is not being used to load template files. "get_template_part()" should be
     |         |     used to load template files.
 163 | WARNING | [ ] Check that require_once is not being used to load template files. "get_template_part()" should be
     |         |     used to load template files.
 170 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 180 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 188 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 188 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 201 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 209 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 209 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 222 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 230 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 230 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 243 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 251 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 251 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 252 | WARNING | [x] Array double arrow not aligned correctly; expected 4 space(s) between "'label'" and double arrow,
     |         |     but found 11.
 253 | WARNING | [x] Array double arrow not aligned correctly; expected 2 space(s) between "'section'" and double
     |         |     arrow, but found 9.
 254 | WARNING | [x] Array double arrow not aligned correctly; expected 1 space(s) between "'settings'" and double
     |         |     arrow, but found 8.
 255 | WARNING | [x] Array double arrow not aligned correctly; expected 1 space(s) between "'priority'" and double
     |         |     arrow, but found 8.
 264 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 272 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 272 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 273 | WARNING | [x] Array double arrow not aligned correctly; expected 4 space(s) between "'label'" and double arrow,
     |         |     but found 11.
 274 | WARNING | [x] Array double arrow not aligned correctly; expected 2 space(s) between "'section'" and double
     |         |     arrow, but found 9.
 275 | WARNING | [x] Array double arrow not aligned correctly; expected 1 space(s) between "'settings'" and double
     |         |     arrow, but found 8.
 276 | WARNING | [x] Array double arrow not aligned correctly; expected 1 space(s) between "'priority'" and double
     |         |     arrow, but found 8.
 283 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 283 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 296 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 304 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 304 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 317 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 325 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 325 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 338 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 346 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 346 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 359 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 370 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 378 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 378 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 391 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 399 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 399 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 412 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 420 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 420 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 433 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 441 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 441 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 454 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 465 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 473 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 473 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 486 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 494 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 494 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 507 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 515 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 515 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 528 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 536 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 536 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 549 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 556 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 564 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 564 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 582 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 589 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 597 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 597 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
-----------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 73 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------


FILE: /Users/nielslange/Development/Themes/storefront/inc/admin/class-storefront-plugin-install.php
-----------------------------------------------------------------------------------------------------------------------
FOUND 4 ERRORS AND 1 WARNING AFFECTING 5 LINES
-----------------------------------------------------------------------------------------------------------------------
  90 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
  91 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 124 | WARNING | [ ] Method name "_is_plugin_installed" should not be prefixed with an underscore to indicate
     |         |     visibility
 135 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 136 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
-----------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 4 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------


FILE: /Users/nielslange/Development/Themes/storefront/inc/admin/class-storefront-admin.php
-----------------------------------------------------------------------------------------------------------------------
FOUND 5 ERRORS AND 7 WARNINGS AFFECTING 12 LINES
-----------------------------------------------------------------------------------------------------------------------
  40 | ERROR   | [ ] Resource version not set in call to wp_enqueue_style(). This means new versions of the style will
     |         |     not always be loaded due to browser caching.
  61 | WARNING | [ ] Check that require_once is not being used to load template files. "get_template_part()" should be
     |         |     used to load template files.
  62 | WARNING | [ ] Check that require_once is not being used to load template files. "get_template_part()" should be
     |         |     used to load template files.
  63 | WARNING | [ ] Check that require_once is not being used to load template files. "get_template_part()" should be
     |         |     used to load template files.
 159 | WARNING | [ ] Check that require_once is not being used to load template files. "get_template_part()" should be
     |         |     used to load template files.
 194 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 195 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 215 | WARNING | [ ] Method name "_is_plugin_installed" should not be prefixed with an underscore to indicate
     |         |     visibility
 226 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 227 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 240 | WARNING | [ ] Check that require_once is not being used to load template files. "get_template_part()" should be
     |         |     used to load template files.
 249 | WARNING | [ ] Check that require_once is not being used to load template files. "get_template_part()" should be
     |         |     used to load template files.
-----------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 4 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------


FILE: /Users/nielslange/Development/Themes/storefront/inc/class-storefront.php
-----------------------------------------------------------------------------------------------------------------------
FOUND 17 ERRORS AFFECTING 16 LINES
-----------------------------------------------------------------------------------------------------------------------
  78 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
  79 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
  93 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 106 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 107 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 124 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 125 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 136 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 137 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 154 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 155 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 189 | ERROR | [x] Opening parenthesis of a multi-line function call must be the last content on the line
 189 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 215 | ERROR | [x] Closing parenthesis of a multi-line function call must be on a line by itself
 326 | ERROR | [ ] Resource version not set in call to wp_enqueue_style(). This means new versions of the style will
     |       |     not always be loaded due to browser caching.
 364 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 519 | ERROR | [ ] Resource version not set in call to wp_enqueue_style(). This means new versions of the style will
     |       |     not always be loaded due to browser caching.
-----------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 15 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------


FILE: /Users/nielslange/Development/Themes/storefront/inc/nux/class-storefront-nux-starter-content.php
-----------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AND 20 WARNINGS AFFECTING 19 LINES
-----------------------------------------------------------------------------------------------------------------------
   42 | WARNING | [ ] Processing form data without nonce verification.
   42 | WARNING | [ ] Processing form data without nonce verification.
  206 | WARNING | [ ] Found precision alignment of 3 spaces.
  272 | WARNING | [ ] Processing form data without nonce verification.
  272 | WARNING | [ ] Processing form data without nonce verification.
  286 | WARNING | [ ] Processing form data without nonce verification.
  286 | WARNING | [ ] Processing form data without nonce verification.
  287 | WARNING | [ ] Processing form data without nonce verification.
  469 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
  645 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
  657 | WARNING | [ ] Passing the $glue and $pieces parameters in reverse order to implode has been deprecated since
      |         |     PHP 7.4; $glue should be the first parameter and $pieces the second
  722 | WARNING | [ ] Method name "_starter_content_products" should not be prefixed with an underscore to indicate
      |         |     visibility
 1035 | WARNING | [ ] Method name "_homepage_blocks_content" should not be prefixed with an underscore to indicate
      |         |     visibility
 1091 | WARNING | [ ] Method name "_replace_homepage_blocks_symbols" should not be prefixed with an underscore to
      |         |     indicate visibility
 1158 | WARNING | [ ] Method name "_get_created_starter_content_products" should not be prefixed with an underscore to
      |         |     indicate visibility
 1180 | WARNING | [ ] Method name "_get_existing_wc_products" should not be prefixed with an underscore to indicate
      |         |     visibility
 1185 | WARNING | [ ] Disabling pagination is not advised, avoid setting `posts_per_page` to `-1`
 1206 | WARNING | [ ] Method name "_query_starter_content" should not be prefixed with an underscore to indicate
      |         |     visibility
 1210 | WARNING | [ ] Disabling pagination is not advised, avoid setting `posts_per_page` to `-1`
 1211 | WARNING | [ ] Detected usage of meta_query, possible slow query.
 1246 | WARNING | [ ] Method name "_get_category_image_attachment_id" should not be prefixed with an underscore to
      |         |     indicate visibility
 1263 | WARNING | [ ] Method name "_validate_tasks" should not be prefixed with an underscore to indicate visibility
-----------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------


FILE: /Users/nielslange/Development/Themes/storefront/inc/nux/class-storefront-nux-admin.php
-----------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AND 3 WARNINGS AFFECTING 4 LINES
-----------------------------------------------------------------------------------------------------------------------
 214 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
 271 | WARNING | [ ] Method name "_set_woocommerce_pages_full_width" should not be prefixed with an underscore to
     |         |     indicate visibility
 287 | WARNING | [ ] Method name "_assign_page_template" should not be prefixed with an underscore to indicate
     |         |     visibility
 301 | WARNING | [ ] Method name "_is_woocommerce_empty" should not be prefixed with an underscore to indicate
     |         |     visibility
-----------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------


FILE: /Users/nielslange/Development/Themes/storefront/inc/jetpack/class-storefront-jetpack.php
----------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------------------------------
 40 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
 41 | ERROR | [x] Only one argument is allowed per line in a multi-line function call
----------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------------------------------


FILE: /Users/nielslange/Development/Themes/storefront/comments.php
-----------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AND 1 WARNING AFFECTING 2 LINES
-----------------------------------------------------------------------------------------------------------------------
 28 | WARNING | [ ] Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native "phpcs:ignore
    |         |     Standard.Category.SniffName.ErrorCode" annotations instead. Found: // WPCS: XSS OK.
    |         |     
 75 | ERROR   | [x] Only one argument is allowed per line in a multi-line function call
-----------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------

Time: 11.36 secs; Memory: 32MB
```

Running `phpcbf . --standard=WordPress,WPThemeReview --ignore="/vendor/*,/node_modules/*" --extensions=php` then leads to:

```
PHPCBF RESULT SUMMARY
-----------------------------------------------------------------------------------------------------------------------
FILE                                                                                                   FIXED  REMAINING
-----------------------------------------------------------------------------------------------------------------------
/Users/nielslange/Development/Themes/storefront/404.php                                                2      0
/Users/nielslange/Development/Themes/storefront/inc/wordpress-shims.php                                11     0
...elslange/Development/Themes/storefront/inc/woocommerce/class-storefront-woocommerce-customizer.php  5      0
...slange/Development/Themes/storefront/inc/woocommerce/storefront-woocommerce-template-functions.php  27     0
/Users/nielslange/Development/Themes/storefront/inc/woocommerce/class-storefront-woocommerce.php       3      3
/Users/nielslange/Development/Themes/storefront/inc/storefront-template-functions.php                  3      3
/Users/nielslange/Development/Themes/storefront/inc/customizer/class-storefront-customizer.php         73     3
/Users/nielslange/Development/Themes/storefront/inc/admin/class-storefront-plugin-install.php          4      1
/Users/nielslange/Development/Themes/storefront/inc/admin/class-storefront-admin.php                   4      8
/Users/nielslange/Development/Themes/storefront/inc/class-storefront.php                               15     2
/Users/nielslange/Development/Themes/storefront/inc/nux/class-storefront-nux-starter-content.php       2      20
/Users/nielslange/Development/Themes/storefront/inc/nux/class-storefront-nux-admin.php                 1      3
/Users/nielslange/Development/Themes/storefront/inc/jetpack/class-storefront-jetpack.php               2      0
/Users/nielslange/Development/Themes/storefront/comments.php                                           1      0
-----------------------------------------------------------------------------------------------------------------------
A TOTAL OF 153 ERRORS WERE FIXED IN 14 FILES
-----------------------------------------------------------------------------------------------------------------------

Time: 48.23 secs; Memory: 34MB
```

### PR DESCRIPTION
**Pull request by haszari**
Prompted by #1393
Fixes #1356

Update: woah cool there is an issue for this! #1356 🎩 @nielslange 

The original goal of this PR was to update the WordPress coding standards PHPcs ruleset, and anything necessary related to it.

I discovered that our phpcs setup wasn't in working order, so have upgraded and replaced various packages. 

~~Work in progress - need to ensure we have enough rules in place that match the existing code. Currently have lots of violations.~~ All violations now fixed and pre-commit should catch issues in new code.

Summary of what's included in this PR:

- Updated and set versions for various phpcs related packages. Key outcome here is getting on the latest WordPress standards. fac37e55bfb523c07e37299edf46946839290d5f
- Added/tweaked commands in composer.json and package.json for visibility and to encourage use/familiarity.
- Set min PHP version of sniffer config to match plugin min PHP version.
- Mega commit fixing all formatting issues automatically using `phpcbf`. 7c90d2311b0433ee9cb0f9d896704c64db3ae0e5
- Manual fixing of a few more interesting violations (~10 issues). One commit per file for visibility. Note this included:
  - Adding/fixing version param in all uses of `wp_enqueue_style`. This means that various third-party assets will be cached less aggressively (now tied to `$storefront_version`).
  - Missing escaping of comment dates in `storefront_comment()` (🧤)
  - Use `gmdate()` instead of ambiguous `date()`.
- Fixed one rule ignore comment to latest format. (Still ignoring, didn't investigate why)

I've fixed all errors but have not touched any warnings. Warnings should be _carefully_ handled as follow up - linter warnings can be very subtle/tricky/risky  🐲 to fix, so I'm not rushing to fix them all in this PR!

**Update** I have bulk-fixed an older style of ignore comment to reduce the amount of open warnings. These comments opt-out of XSS checks, and we have a lot of them. c2b82fac62038925df1db0bf0fb6f7ed555808c9

- Old comment `// WPCS: XSS ok.`
- Current comment, using `phpcs:ignore` `// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped`

##### Travis / CI is _not_ included (yet)
Note there's nothing in this PR to trigger linter check as a CI step (e.g. on PRs). We should do that next! #1395 

#### Test linter tooling
1. Get on this branch, `composer install`, `npm install`.
   - Note: `composer install` is what sets up the `pre-commit` hook (TIL).
2. Run linter `npm run lint:php`. Should pass with flying colours (* and plenty of warnings 😿).
3. Write some [bad PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/).
4. Run linter again - should tell you what's wrong.
5. Try to commit the bad changes - commit hook should catch it. 
5. Write some PHP with auto-fixable changes (e.g. spaces/tabs, dangling commas, long lines, multi-line func calls).
6. Run auto fix to `npm run lint:php:fix`. Should fix the errors so you can commit.

#### Broad smoke testing of a cross-section of Storefront functionality
As this is a big change, any testing of Storefront features would be appreciated. 

cc @woocommerce/rubik I've tagged you all for review – first in first served. I don't expect everyone to review – grateful for any reviews or testing.

If you could run this PR (or master after merge) in your dev environment and keep an eye out for issues I would really appreciate it 🙌 🙇 

#### Affected flows
- Everything (in theory). 
- Comment dates now escaped. 

### Changelog

> Dev – Reinstate phpcs PHP linting. #1463

### CODE REVIEW

File: phpcs.xml [hunk 1]
```diff
@@ -21,27 +21,6 @@
 		<exclude name="PHPCompatibility.PHP.NewKeywords.t_namespaceFound" />
 	</rule>
 
-	<rule ref="WordPress">
```
Comment (Aljullu) [inline phpcs.xml:24]: I'm missing why this rule was removed. Is it not adding anything? I tried re-enabling it locally to test it and doesn't seem to make any difference, but wanted to ask to be sure.

**Review by Aljullu — APPROVED** [review]
Nice work with this! Really great to see projects getting into a shared set of linting rules and code styles. I just left a comment below about a rule that was removed, but everything else looking and testing great. :ok_hand: 

@haszari FYI I updated the testing instructions in the PR description. They were pointing to `npm run php:lint` as the command to run, but it should be `npm run lint:php`.

Also notice that when I run that command I get several warnings. I assume that's expected?

File: phpcs.xml (hunk 1 above)
Comment (haszari) [reply phpcs.xml:24]: That was unintentional … I'll put it back. I was hacking on this xml file heavily trying to confirm the right rulesets were applying, thanks for catching.

**Comment by haszari** [comment]
I've done some broad smoke testing and didn't see any issues - so merging. Please all keep an eye out for any impact of this change (🤞).

**Pull request merged**